Sources/Overload/OvEditor/include/OvEditor/Core/EditorActions.h

```diff
@@ -166,38 +166,19 @@ namespace OvEditor::Core
 		* Create an empty actor
 		* @param p_focusOnCreation
 		* @param p_parent
+        * @param p_name
 		*/
-		OvCore::ECS::Actor&	CreateEmptyActor(bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr);
+		OvCore::ECS::Actor&	CreateEmptyActor(bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr, const std::string& p_name = "");
 
 		/**
 		* Create an actor with a model renderer and a material renderer. The model renderer with use the model identified
 		* by the given path
 		* @param p_path
 		* @param p_focusOnCreation
 		* @param p_parent
+        * @param p_name
 		*/
-		OvCore::ECS::Actor&	CreateActorWithModel(const std::string& p_path, bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr);
-
-		/**
-		* Create an actor whith a physical box
-		* @param p_focusOnCreation
-		* @param p_parent
-		*/
-		OvCore::ECS::Actor&	CreatePhysicalBox(bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr);
-
-		/**
-		* Create an actor whith a physical sphere
-		* @param p_focusOnCreation
-		* @param p_parent
-		*/
-		OvCore::ECS::Actor&	CreatePhysicalSphere(bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr);
-
-		/**
-		* Create an actor whith a physical capsule
-		* @param p_focusOnCreation
-		* @param p_parent
-		*/
-		OvCore::ECS::Actor& CreatePhysicalCapsule(bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr);
+		OvCore::ECS::Actor&	CreateActorWithModel(const std::string& p_path, bool p_focusOnCreation = true, OvCore::ECS::Actor* p_parent = nullptr, const std::string& p_name = "");
```

Comment:
I saw in `CreateActorWithModel` that both `m_context.modelManager[p_path]` and `m_context.materialManager[":Materials\\Default.ovmat"]` return a `T*` pointer based on `AResourceManager<T>::operator[](const string&)` overload.
We should then consider `auto* const model` and `auto* const material` accordingly instead of `auto model` and `auto material`

Reply:
We can also set `bool p_focusOnCreation` constant